fix: fix data acquisition from TSL2561 luminosity sensor

marcnause opened the pull request:
Fixes #2783

## Changes 
- Try to find sensor by checking all possible sensor IDs (0x29, 0x39, 0x49)
- Fix data acquisition
- Display sensor IDs in hex format in Sensors screen

I think there are still a few things to fix in the screen, but data acquisition works, which was the scope of the issue.

## Screenshots / Recordings 
https://github.com/user-attachments/assets/2c104bca-7251-427d-94c2-fbf6bb2cda40

**Checklist**: <!-- Please tick following check boxes with `[x]` if the respective task is completed -->
- [x] **No hard coding**: I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard coding any value.
- [x] **No end of file edits**: No modifications done at end of resource files `strings.xml`, `dimens.xml` or `colors.xml`.
- [x] **Code reformatting**: I have reformatted code and fixed indentation in every file included in this pull request.
- [x] **No extra space**: My code does not contain any extra lines or extra spaces than the ones that are necessary.

## Summary by Sourcery

Add support for detecting TSL2561 sensors at multiple I2C addresses and correct raw luminosity reads, while updating the sensor screen to display addresses in hex

New Features:
- Detect TSL2561 devices by probing all potential I2C addresses (0x29, 0x39, 0x49)
- Display sensor addresses in hexadecimal format in the Sensor screen

Bug Fixes:
- Fix raw data acquisition for full spectrum, infrared, and visible channels using correct read commands and bit operations

Enhancements:
- Refactor TSL2561 class to centralize constants, use byte types, and streamline enable/read logic
- Update SensorActivity to use integer lists for addresses and rebuild scan output accordingly

marcnause committed:
fix: fix data acquisition from TSL2561 luminosity sensor

Fixes https://github.com/fossasia/pslab-android/issues/2783

marcnause committed:
refactor: remove unused code

AsCress commented on app/src/main/java/io/pslab/communication/sensors/TSL2561.java:
```diff
@@ -2,123 +2,130 @@
+             * If a different value is returned, we can check for teh expected value (see data sheet
+             * for details).
+             */
+            for (byte addr : ADDRESSES) {
```
@marcnause Isn't this handled before in the sensor auto scan ? If the auto scan detects the `TSL2561`, it would show up in the list, which would make this screen accessible to the user. By the time, the user reaches this screen, would it even be possible that the TSL2561 isn't found ?
Is this for handling the conflicting addresses supported by TSL2561 (`0x29` and `0x39`), cause as far as I remember in my discussion with @bessman, there's no way of finding out which sensor is present on a conflicting address rather than trying to read data from it. So, if we're going ahead on that path, do you think that we should implement this check in the higher layers itself (in auto scan), to rule out the showing up of the sensors not present itself ? What are your views on this ?

marcnause replied:
I don't know if all or at least most sensors have a way of querying an ID or any other distinct value which lets us decide which sensor is present for a certain address. If this was the case, then I think it would make sense to add some intelligence to the auto scan.

I decided to put the detection into the code for the TSL2561 since currently we don't forward the address of a sensor from the Sensor**s** screen to the sensor specific screens. I also did not want to put too much effort into the legacy version, I only wanted to provide a template for the Flutter version which is at least doing the bare minimum of what is required. I only stumbled over the different possible addresses since the sensor I bought does not use the default address.

AsCress replied:
@marcnause Makes sense. We'll take a call on where to implement these checks in the flutter app. You're right, for the Java app this is fine as it just as to act as a template for the flutter port.

AsCress reviewed (commented):
@marcnause Thank you very much for this !

marcnause committed:
fix: sensor was only probed successfully once

When the sensor was probed a second time after being active, the expected value was not returned and the app did not recognize the sensor.
